### PR TITLE
Switch OLED addressing mode to contiguous.

### PR DESCRIPTION
**Pull request by cnlohr**

### CODE REVIEW

File: firmware/user/display/oled.h [hunk 1]
```diff
@@ -35,6 +35,7 @@ bool initOLED(bool reset);
 void drawPixel(int16_t x, int16_t y, color c);
 color getPixel(int16_t x, int16_t y);
 bool ICACHE_FLASH_ATTR setOLEDparams(bool turnOnOff);
+int ICACHE_FLASH_ATTR updateOLEDScreenRange( uint8_t minX, uint8_t maxX, uint8_t minPage, uint8_t maxPage );
```
Comment (AEFeinstein) [inline firmware/user/display/oled.h:38]: `updateOLED()` is already called on a timer. If you're exposing `updateOLEDScreenRange()` are you planning on a way to turn off the periodic `updateOLED()` calls?

File: firmware/user/display/oled.c [hunk 2]
```diff
@@ -322,70 +303,35 @@ bool ICACHE_FLASH_ATTR setOLEDparams(bool turnOnOff)
...
+		uint8_t * prior = &priorFb[index];
+		uint8_t * cur = &currentFb[index];
+
+        for( page = minPage; page <= maxPage; page++ )
         {
-            // Mark it
-            bounds[0] = col;
-            // Then look for the last difference
-            for (col = SSD1306_NUM_COLS - 1; col >= 0; col--)
+            uint8_t c = *(prior++) = *(cur++);
```
Comment (AEFeinstein) [inline firmware/user/display/oled.c]: Sneaky memcpy

File: firmware/user/display/oled.c [hunk 3]
```diff
@@ -322,70 +303,35 @@ bool ICACHE_FLASH_ATTR setOLEDparams(bool turnOnOff)
...
-    bool anyDiffs = false;
-    // Look for the first difference
-    for (col = 0; col < SSD1306_NUM_COLS; col++)
+    uint8_t encountered_error = false;
+    uint8_t x, page;
+
+    setColumnAddress( minX, maxX );
+    setPageAddress( minPage, maxPage );
+
+    if( cnlohr_i2c_end_transaction() ) encountered_error = true;
```
Comment (AEFeinstein) [inline firmware/user/display/oled.c:314]: Where is the equivalent `cnlohr_i2c_start_transaction()`? The only one I see in `setOLEDparams()`

Comment (AEFeinstein) [reply firmware/user/display/oled.c:314]: Oh, you're calling `SendStart()` directly. Is that really necessary?

File: firmware/user/display/oled.h (hunk 1 above)
Comment (cnlohr) [reply firmware/user/display/oled.h:38]: I was not planning on it - should I?  I just figured I'd let the rest of the system do its thing.  this just explicitly schleps the pixels to the display.

File: firmware/user/display/oled.c (hunk 3 above)
Comment (cnlohr) [reply firmware/user/display/oled.c:314]: Look a few lines up.  `cnlohr_i2c_start_transaction` is there.

**Comment by cnlohr** [comment]
Ok, fixed the merge.

File: firmware/user/display/oled.h (hunk 1 above)
Comment (AEFeinstein) [reply firmware/user/display/oled.h:38]: No, the behavior is fine as-is. I saw the declaration in a header, so I thought you were planning on using it outside `oled.c`. It's only used in there now, so the declaration could be in `oled.c` to keep scope a little tighter.

File: firmware/user/display/oled.c [hunk 4]
```diff
@@ -139,7 +139,7 @@ bool fbChanges = false;
  */
 void ICACHE_FLASH_ATTR clearDisplay(void)
 {
-    ets_memset(currentFb, 0, sizeof(currentFb));
+    ets_memset(currentFb, 0, (OLED_WIDTH * (OLED_HEIGHT / 8)) );
```
Comment (AEFeinstein) [inline firmware/user/display/oled.c:142]: 🍠🥔

File: firmware/user/display/oled.c [hunk 5]
```diff
@@ -322,70 +303,31 @@ bool ICACHE_FLASH_ATTR setOLEDparams(bool turnOnOff)
...
-    bool anyDiffs = false;
-    // Look for the first difference
-    for (col = 0; col < SSD1306_NUM_COLS; col++)
+    uint8_t encountered_error = false;
+    uint8_t x, page;
+
+    setColumnAddress( minX, maxX );
+    setPageAddress( minPage, maxPage );
+
+    if( cnlohr_i2c_end_transaction() ) encountered_error = true;
```
Comment (AEFeinstein) [inline firmware/user/display/oled.c:314]: I think `cnlohr_i2c_start_transaction()` disappeared again...

File: firmware/user/display/oled.c (hunk 4 above)
Comment (cnlohr) [reply firmware/user/display/oled.c:142]: Sorry, this was left over from when I was actually using pointers to deal with the buffers.  I flipped back.

File: firmware/user/display/oled.c (hunk 5 above)
Comment (AEFeinstein) [reply firmware/user/display/oled.c:314]: Seems like you removed it in https://github.com/AEFeinstein/Super-2021-Swadge-FW-Sandbox/pull/48/commits/c6b641033aabf6b25704c4e78aa38fc40986a887

File: firmware/user/display/oled.h (hunk 1 above)
Comment (cnlohr) [reply firmware/user/display/oled.h:38]: I think it may make sense to expose to folks.  Like the games know what part of the screen they're updating.

**Pull request closed without merging**